From 7c4788726ee1b0bb333319b806ddb662a8572a37 Mon Sep 17 00:00:00 2001
From: Ansuel Smith <ansuelsmth@gmail.com>
Date: Sat, 14 May 2022 21:26:59 +0200
Subject: [PATCH 12/12] ecm_interface: rework br_dev_update_stats to internal
 function

Rework br_dev_update_stats to an internal function instead of adding a
symbol in the kernel.

Signed-off-by: Ansuel Smith <ansuelsmth@gmail.com>
---
 ecm_interface.c | 21 ++++++++++++++++++++-
 1 file changed, 20 insertions(+), 1 deletion(-)

--- a/ecm_interface.c
+++ b/ecm_interface.c
@@ -741,6 +741,25 @@ bool ecm_interface_mac_addr_get(ip_addr_
 }
 EXPORT_SYMBOL(ecm_interface_mac_addr_get);
 
+static void ecm_interface_br_dev_update_stats(struct net_device *dev,
+					      struct rtnl_link_stats64 *nlstats)
+{
+	struct pcpu_sw_netstats *tstats;
+
+	/* Is this a bridge? */
+	if (!(dev->priv_flags & IFF_EBRIDGE))
+		return;
+
+	tstats = this_cpu_ptr(dev->tstats);
+
+	u64_stats_update_begin(&tstats->syncp);
+	tstats->rx_packets += nlstats->rx_packets;
+	tstats->rx_bytes += nlstats->rx_bytes;
+	tstats->tx_packets += nlstats->tx_packets;
+	tstats->tx_bytes += nlstats->tx_bytes;
+	u64_stats_update_end(&tstats->syncp);
+}
+
 #ifdef ECM_IPV6_ENABLE
 /* ecm_interface_ipv6_dev_find_and_hold()
  *	Find (and hold) net device that has the given address.
@@ -6577,7 +6596,7 @@ skip_bridge_refresh:
 			stats.rx_bytes = rx_bytes;
 			stats.tx_packets = tx_packets;
 			stats.tx_bytes = tx_bytes;
-			br_dev_update_stats(dev, &stats);
+			ecm_interface_br_dev_update_stats(dev, &stats);
 			dev_put(dev);
 			continue;
 
